controller/impl.go

```diff
@@ -99,9 +107,6 @@ func (f *Factory) runCommandLoop(ctx context.Context, logger *zerolog.Logger, po
 	case err = <-errCh:
```

Comment:
If you use an `errGroup` we won't need this `select` anymore. It would simply be an `errGroup.Wait()`.